dvc/config_schema.py

```diff
@@ -179,7 +179,12 @@ class RelPath(str):
                     "webhdfs_alias": str,
                     **REMOTE_COMMON,
                 },
-                "azure": {"connection_string": str, **REMOTE_COMMON},
+                "azure": {
+                    "connection_string": str,
```

Comment:
- we want to support Azure ServicePrincipal credentials on the DVC config level (tenant_id, client_id, client_secret) - should we do this as part of this PR, or keep it purely as a migration one?

Reply:
We talked about this with @efiop, I didn't have chance to test it personally today but after verifying it works I'll add it tomorrow since it is already supported in adlfs
https://github.com/dask/adlfs/blob/2bd9d9302e5d3511ce636246cbd2d7770000f00c/adlfs/spec.py#L304-L309